// react-weui/lib/components/mediabox/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _mediabox = require('./mediabox');

var _mediabox2 = _interopRequireDefault(_mediabox);

var _mediabox_header = require('./mediabox_header');

var _mediabox_header2 = _interopRequireDefault(_mediabox_header);

var _mediabox_body = require('./mediabox_body');

var _mediabox_body2 = _interopRequireDefault(_mediabox_body);

var _mediabox_title = require('./mediabox_title');

var _mediabox_title2 = _interopRequireDefault(_mediabox_title);

var _mediabox_desc = require('./mediabox_desc');

var _mediabox_desc2 = _interopRequireDefault(_mediabox_desc);

var _mediabox_info = require('./mediabox_info');

var _mediabox_info2 = _interopRequireDefault(_mediabox_info);

var _mediabox_info_meta = require('./mediabox_info_meta');

var _mediabox_info_meta2 = _interopRequireDefault(_mediabox_info_meta);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
    MediaBox: _mediabox2.default,
    MediaBoxHeader: _mediabox_header2.default,
    MediaBoxBody: _mediabox_body2.default,
    MediaBoxTitle: _mediabox_title2.default,
    MediaBoxDescription: _mediabox_desc2.default,
    MediaBoxInfo: _mediabox_info2.default,
    MediaBoxInfoMeta: _mediabox_info_meta2.default
}; /**
    * Created by n7best
    */

module.exports = exports['default'];
